src/core/push/safari.ts

```diff
@@ -46,7 +46,43 @@ export default class SafariPushManager implements PushOpsManager {
         return result;
     }
 
+    async attemptPushRegister(ctx: Context): Promise<void> {
+        const wasUnregistered = await get<boolean>('wasUnregistered');
+
+        if (wasUnregistered) {
+            console.info('Was unregistered before, not calling push register');
+            return;
+        }
+
+        return this.pushRegister(ctx);
+    }
+
+    async pushUnregister(ctx: Context): Promise<void> {
+        await set<boolean>('wasUnregistered', true);
```

Comment:
Should we set this after emitting the event?

If tracking the unregister event fails, we won't be unregistered, and the web SDK doesn't cache events atm, so a failure could result in inconsistent state.

Reply:
Changed